### PR TITLE
feat: Add ability to specify MTU when creating Network

### LINKED ISSUE #1246: Support MTU change for VPC
/kind feature

**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]

As of now, CAPG doesn't allow to set MTU for VPC which creates a VPC with default MTU as 1460. So option needs to be available to change MTU to different value. 

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

### PR DESCRIPTION
**Pull request by tasdikrahman**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

User can set `GCPCluster.Spec.Network.Mtu` for setting the MTU for the network being created. If nothing is passed the MTU will default to 1440 which is the default set here https://cloud.google.com/vpc/docs/mtu

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1246

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
feat: Add ability to add GCPCluster.Spec.Network.Mtu 
```

### CODE REVIEW

**Review by cpanato — COMMENTED** [review]
/ok-to-test

**Comment by tasdikrahman** [comment]
Thanks for the review @cpanato , let me know if you would like me to add any more changes on this PR before approval. Thank you.

**Comment by cpanato** [comment]
thanks

/approve

/assign @damdo

File: api/v1beta1/types.go
Comment (damdo) [inline api/v1beta1/types.go:144]: I think we should add openAPI validation for this field only allowing the two values you are mentioning.
Also where are those values coming from? We might want to leave a message mentioning that.

Comment (damdo) [reply api/v1beta1/types.go:144]: Also we could add a validation step in the validation code (e.g. webooks).
And a couple of unit test checking that is validated correctly.

**Review by damdo — CHANGES_REQUESTED** [review]

**Comment by damdo** [comment]
/hold
For changes to be added

File: api/v1beta1/types.go
Comment (tasdikrahman) [reply api/v1beta1/types.go:144]: > I think we should add openAPI validation for this field only allowing the two values you are mentioning.
Also where are those values coming from? We might want to leave a message mentioning that.

Hello there, I fixed the validation code to have the minimum and maximum values present. The values of minimum of 1300 and maximum of 8896 are taken from https://pkg.go.dev/google.golang.org/api/compute/v1#Network

> Also we could add a validation step in the validation code (e.g. webooks).
And a couple of unit test checking that is validated correctly.

Will add some unit tests exercising this.

**Comment by damdo** [comment]
Thank you for implementing the requested changes @tasdikrahman 
/lgtm
/unhold

**Pull request merged**
